File: app/src/main/java/com/weathercock/android_cleanarchitecture/data/repository/AboutRepositoryImpl.java
package com.weathercock.android_cleanarchitecture.data.repository;

import android.content.Context;

import com.weathercock.android_cleanarchitecture.R;
import com.weathercock.android_cleanarchitecture.data.util.StoreUtils;
import com.weathercock.android_cleanarchitecture.data.util.VersionUtils;
import com.weathercock.android_cleanarchitecture.domain.repository.AboutRepository;

/**
 * About Repository Implement
 * <p>
 * Created by tamura_k on 2016/05/27.
 */
public class AboutRepositoryImpl implements AboutRepository {

    private Context context;

    public AboutRepositoryImpl(Context context) {
        this.context = context;
    }

    @Override
    public String getPlayStoreUrl() {
        return StoreUtils.getStoreAppLink(context);
    }

    @Override
    public String getMailUrl() {
        return context.getString(R.string.developer_mail_address);
    }

    @Override
    public String getWebSiteUrl() {
        return context.getString(R.string.developer_website_url);
    }

    @Override
    public String getCurrentVersion() {
        return VersionUtils.getVersionName(context);
    }

}

File: data/src/main/java/com/kazakago/cleanarchitecture/data/repository/about/AboutRepositoryImpl.kt
package com.kazakago.cleanarchitecture.data.repository.about

import android.content.Context
import com.kazakago.cleanarchitecture.data.R
import com.kazakago.cleanarchitecture.data.util.StoreUtils
import com.kazakago.cleanarchitecture.data.util.VersionUtils
import com.kazakago.cleanarchitecture.domain.model.about.AppInfo
import com.kazakago.cleanarchitecture.domain.model.about.DeveloperInfo
import com.kazakago.cleanarchitecture.domain.repository.about.AboutRepository
import java.net.URL

class AboutRepositoryImpl(private val context: Context) : AboutRepository {

    override fun getAppInfo(): AppInfo {
        return AppInfo(VersionUtils.getVersionName(context),
                VersionUtils.getVersionCode(context),
                StoreUtils.getStoreAppLink(context))
    }

    override fun getDeveloperInfo(): DeveloperInfo {
        return DeveloperInfo(context.getString(R.string.developer_name),
                context.getString(R.string.developer_mail_address),
                URL(context.getString(R.string.developer_website_url)))
    }

}
